templates/diskovc/diskovc.py

```diff
@@ -0,0 +1,162 @@
+        ovcs = self.api.services.find(template_uid=self.OVC_TEMPLATE, name=data.get('openvcloud', None))
+        if len(ovcs) != 1:
+            raise RuntimeError('found %s openvcloud connections, requires exactly 1' % len(ovcs))
+        data['openvcloud'] = ovcs[0].name
+
+        # ensure location is given
+        if not data['location']:
+            raise j.exceptions.Input("location is not given")      
+
+        # ensure uploaded key
+        self.sshkey
+
+        # ensure that disk has a valid type
+        if data['type']  and data['type'].upper() not in ["D", "B"]:
+            raise j.exceptions.Input("diskovc's type must be data (D) or boot (B) only")
+
+        # ensure that limits are given correctly
+        if (data['maxIOPS'] or data['totalIopsSec']) and (data['readIopsSec'] or data['writeIopsSec']):
+            raise j.exceptions.Input("total and read/write of iops_sec cannot be set at the same time")
+
+        if data['totalBytesSec'] and (data['readBytesSec'] or data['writeBytesSec']):
+            raise j.exceptions.Input("total and read/write of bytes_sec cannot be set at the same time")
+
+        if data['totalBytesSecMax'] and (data['readBytesSecMax'] or data.writeBytesSecMax):
+            raise j.exceptions.Input("total and read/write of bytes_sec_max cannot be set at the same time")
+
+        if data['totalIopsSecMax'] and (data['readIopsSecMax'] or data['writeIopsSecMax']):
+            raise j.exceptions.Input("total and read/write of iops_sec_max cannot be set at the same time")
+
+
+    def update_data(self, data):
+        # merge new data
+        self.data.update(data)
+        
+        self.save()
+
+    @property
+    def sshkey(self):
+        """ Get a path and keyname of the sshkey service """
+
+        sshkeys = self.api.services.find(template_uid=self.SSH_TEMPLATE)
+        if len(sshkeys) != 1:
+            raise RuntimeError('found %s ssh services, requires exactly 1' % len(sshkeys))
+
+        # Get key name and path
+        path = sshkeys[0].data['path']
+        key = path.split('/')[-1]
+  
+        return key 
+
+    @property
+    def ovc(self):
+        """
+        An ovc connection instance
+        """
+        if self._ovc is not None:
+            return self._ovc
+
+        self._ovc = j.clients.openvcloud.get(self.data['openvcloud'])
+        return self._ovc        
+
+    @property
+    def account(self):
+        """
+        Return an account
+        """
+        if self._account is not None:
+            return self._account
+
+        # Get object for an OVC service, make sure exactly one is running
+        accounts = self.api.services.find(template_uid=self.ACCOUNT_TEMPLATE, name=self.data.get('account', None))
+        if len(accounts) != 1:
+            raise RuntimeError('found %s openvcloud connections, requires exactly 1' % len(accounts))
+        account = accounts[0].name  
+        self._account = self.ovc.account_get(account, create=True)
```

Comment:
create=True? I don't think you should create any account from this template. The account should already exists when you reach this point